backend/init_data/skills/browser/_base.py

```
@@ -0,0 +1,375 @@
+
+    This class manages browser lifecycle within a sandbox session:
+    - Browser creation and reuse
+    - Page management
+    - Automatic cleanup
+
+    Browser instances are reused within the same sandbox session to maintain
+    state (cookies, localStorage, etc.) across multiple tool calls.
+    """
+
+    # Class-level dictionary to store singleton instances per task_id
+    _instances: dict[int, "BrowserSessionManager"] = {}
```

Ruff (0.14.14)

[warning] Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)